### PR TITLE
fix(workflow): declare only through the reason

### LINKED ISSUE #3138: fix(workflow): stop prose matching deciding PR recovery
## Problem

PR recovery decides whether an agent's work already landed by searching the agent's free-text output for substrings. One signal matches the bare word "main" anywhere in that prose, so an agent that merely mentions the main branch reads as evidence its change is already on it. Three more signal sets in the same file are built the same way.

A false positive closes a task as landed when nothing shipped. A false negative parks a task on a human that needed no human. The provider layer hit this class twice in the last week and moved to a classification value the step returns instead of matching prose.

## Expected outcome

- An agent mentioning the default branch in passing does not register as evidence that its work already landed
- The recovery decision reads a value the step returns or the repository's own state, never a substring of prose
- Each recovery outcome has a test pinning the input that produces it, covering already landed, no pull request needed, and retry
- A recovery run that cannot classify its input parks the task with a reason instead of guessing
- Tasks that recovery already resolves correctly today keep resolving the same way

### PR DESCRIPTION
**Pull request by Automaat**
## Motivation

The already-fixed-on-main recovery closed a task as `done` based on a substring scan of whatever prose the run emitted. Its base-branch signal was an unanchored `main`, which matches `main.go`, `func main()`, `remaining`, `maintain` and `domain`, so it was true for almost any sentence and the phrase set was the only real gate. The scan had no notion of negation either: a run reporting that it had checked and the change was **not** on main satisfied both halves and closed the task.

The repo-state proof behind it does not save the situation. No commits ahead, a clean tree, and HEAD reachable from the origin base are all trivially true for a branch the agent never committed to, which is exactly the state a run that gave up leaves behind. Text was the only thing standing between "the agent did nothing" and "task closed as done". #3138.

> Changelog: fix(workflow): declare only through the reason

## Implementation information

A task now closes as an already-landed duplicate only when its entire status reason is a JSON object declaring it, which the agent sets with a deliberate `sybra-cli update --status-reason` call. Nothing is extracted from a larger body of text and the run's response is not a channel at all. The repo-state proof still has to hold on top of the declaration, and a run that declares nothing, or declares something unreadable, stays parked with the reason recorded on the task rather than only in the log.

The prose heuristic is deleted rather than narrowed, and so is every intermediate design this PR went through. That history is the argument for the final shape, so it is worth stating plainly: word-anchoring the base-branch signal still closed on `"I checked whether this was already fixed on main; it is NOT"`; accepting a fenced `sybra-recovery` block still closed when an agent echoed a filled-in block from the task body and disclaimed it in prose; accepting a whole-string object from the run's response still closed a task whose status reason explicitly refused to declare one. Each fix removed one shape and the class survived. Requiring the whole status reason removes the class, because a quotation inside prose is never the whole string and narration cannot invoke the CLI.

The declared reason is bounded with `textutil` before it reaches task frontmatter. A 20,000-character reason previously landed verbatim in the task file.

## Supporting documentation

**Behaviour change worth calling out.** The issue asks both for "never a substring of prose" and for tasks that recovery resolves correctly today to keep resolving. Those conflict and this PR takes the first. Until the new prompt reaches running agents, a duplicate task that would previously have auto-closed on prose now parks for a human. That is the issue's own stated preference for a run it cannot classify, and parking a real duplicate is the cheaper error than closing a task where nothing shipped.

`prompt-lab-author.yaml` also runs `role: implementation` without the declaration contract, so its runs no longer auto-recover. Left deliberately: that workflow authors prompts rather than landing changes.

**Verification.** Every finding above came from adversarial review and from adversarial manual testing against a real `sybra-server` in an isolated `SYBRA_HOME` with fake provider CLIs, not from reading the diff. The final run drove 39 tasks through the recovery; exactly two reached `done`, both intended. The eight response-channel false closes from the previous run are fixed, and a declaration on a branch that does have commits still parks.

Coverage is mutation-checked in both directions. Restoring the prose scan turns `TestAdvanceStep_AlreadyFixedOnMainUndeclaredStaysHumanRequired` red on all five inputs; restoring the response text as a channel turns `TestAdvanceStep_AlreadyFixedOnMainResponseTextIsNotAChannel` red. Before this PR the equivalent engine tests survived the first mutation, which is why the engine-level pins were added next to the pure-function table.

**One thing found and deliberately not fixed here.** `route_pr_fix_result` writes a pr-fix agent's `SYBRA_PR_FIX_REASON` sentinel verbatim as the whole status reason, so an agent can get a syntactically valid declaration into a reason without the CLI call. Testing confirmed it cannot reach `done` three ways: the recovery requires `role: implementation`, no shipped workflow routes a pr-fix human-required park into an implementation step, and every transition into human-required clears the reason. It is a laundering surface with no current exit, so it belongs in its own issue rather than widening this diff.

Closes #3138

### CODE REVIEW

File: internal/workflow/builtin/simple-task-implement.yaml [hunk 1]
```diff
@@ -114,6 +114,20 @@ steps:
...
         Before finishing, re-check your change against a short checklist:
         tests updated, edge cases covered, no leftover debug code.
+
+        If you made no commits because the requested change is already present
+        on the base branch, so this task is a duplicate that needs no PR,
+        declare it by setting the task's status reason to exactly one JSON
+        object and nothing else:
+
+            sybra-cli update <task-id> --status human-required \
+              --status-reason '{"decision":"already-fixed-on-main","reason":"<one sentence>"}'
```
[the remark this answers is not in the document]
Comment (Automaat) [reply internal/workflow/builtin/simple-task-implement.yaml]: Fixed in aedb1b79 — the example now uses `{{.Task.ID}}`, matching the templating the other builtin prompts use, so a copy/paste of the line is directly runnable.
